suse_migration_services/units/product_setup.py

```diff
@@ -58,29 +60,74 @@ def main():
         products_metadata = os.sep.join(
             [root_path, 'etc', 'products.d']
         )
-        baseproduct = os.sep.join(
-            [products_metadata, 'baseproduct']
-        )
-        if os.path.exists(baseproduct):
+        if os.path.exists(products_metadata):
             log.info('Creating backup of Product data')
             Command.run(
                 [
                     'rsync', '-zav', '--delete', products_metadata + os.sep,
                     '/tmp/products.d.backup/'
                 ]
             )
-            log.info('Updating Base Product to be suitable for migration')
-            xml = ElementTree()
-            xml.parse(baseproduct)
+        baseproduct = get_baseproduct(products_metadata)
+        if baseproduct.count != 1:
+            if not baseproduct.count:
+                message = 'There is no baseproduct'
+                log.error(message)
+                raise DistMigrationProductSetupException(message)
+            else:
+                message = ('Found multiple product definitions'
+                           'without element <flavor>')
+                log.error(message)
+                raise DistMigrationProductSetupException(message)
```

Comment:
This could be shorten a bit

```python
if not baseproduct.count:
    message = 'a'
else:
    message = 'b'
log.error(message)
raise DistMigrationProductSetupException(message)
```